ScoutSuite/providers/aws/facade/s3.py

```diff
@@ -0,0 +1,185 @@
+from asyncio import Lock
+import json
+from botocore.exceptions import ClientError
+
+from ScoutSuite.providers.aws.facade.utils import AWSFacadeUtils
+from ScoutSuite.providers.aws.facade.basefacade import AWSBaseFacade
+from ScoutSuite.providers.utils import run_concurrently
+from ScoutSuite.providers.utils import get_non_provider_id
+from ScoutSuite.core.console import print_error, print_exception
+
+
+class S3Facade(AWSBaseFacade):
+
+    async def get_buckets(self):
+        client = AWSFacadeUtils.get_client('s3', None, self.session)
+        buckets = (await run_concurrently(lambda: client.list_buckets()))['Buckets']
+
+        for bucket in buckets:
```

Comment:
I think this loop should be parallelized, Xavier told us about an AWS account with millions of buckets.. Parallelizing the fetching of all these attributes should be necessary.

Reply:
(I can take care of this one, if you want.)

Reply:
I think I would leave a todo and come back to later. We'll have to do some batching because it wouldn't be smart to start 1 million requests simultaneously. I'm going to try and create a generic equivalent of `run_concurrently` for task lists.

Reply:
@Aboisier sounds good. If it's something you'll leave to the side please open an issue to track it. There's enough forgotten `TODO`s in the code as is!

Reply:
We've been leaving a lot of Todos, we plan to address them all before we merge the refactoring branch, don't worry!